website/docs/en/config/experiments.mdx

```diff
@@ -959,3 +959,16 @@ export default {
   },
 };
 ```
+
+## experiments.lazyBarrel
+
+<ApiMeta addedVersion="1.5.0" />
+
+- **Type:** `boolean`
+- **Default:** `false`
+
+Whether to skip building unused re-export modules in side-effect-free barrel files to optimize build performance.
+
+A module is considered side-effect-free when it is marked with [`"sideEffects": false` in package.json or flagged by module.rule.sideEffects](/config/optimization#optimizationsideeffects).
```

Comment:
If a package is not configured with "sideEffects": false, will Rspack analyze its sideEffects?

Reply:
Will analyze but not available for lazyBarrel, see https://github.com/web-infra-dev/rspack/discussions/11273#:~:text=Why%20depend%20on%20%22sideEffects%22%3A%20false%20in%20package.json%2C%20can%20Rspack%20automatically%20analyze%20a%20module%20is%20side%20effects%20free%20or%20not%3F

Reply:
A FAQ would be helpful👍